sqlitecluster/SQLiteNode.cpp

```diff
@@ -2580,6 +2541,62 @@ void SQLiteNode::_recvSynchronize(Peer* peer, const SData& message) {
         throw "commits remaining at end";
 }
 
+SQLiteNode::Peer* SQLiteNode::_updateSyncPeer()
+{
+    Peer* newSyncPeer = 0;
+    uint64_t commitCount = _db.getCommitCount();
+    for (auto peer : peerList) {
+        // If any of these conditions are true, then we can't use this peer.
+        if (peer->test("Permaslave") ||                 // Peer is a permaslave
+            !peer->test("LoggedIn") ||                  // Peer is not logged in.
+            !SIEquals((*peer)["State"], "SLAVING") ||   // Peer is not slaving.
```

Comment:
I don't think it needs to be SLAVING -- it's ok to synchronize from a non-slaving peer, such as if everybody is just sitting around waiting for the master to stand up.